app/src/main/java/me/ccrama/redditslide/Adapters/SubChooseAdapter.java

```diff
@@ -67,12 +67,23 @@ public Filter getFilter() {
         return filter;
     }
 
-    @Override
-    public View getView(final int position, View convertView, ViewGroup parent) {
-        convertView = LayoutInflater.from(getContext()).inflate(R.layout.subforsublist, parent, false);
+    private static class ViewHolderItem {
+		private TextView t;
+	}
 
-        final TextView t =
-                ((TextView) convertView.findViewById(R.id.name));
+	@Override
+    public View getView(final int position, View convertView, ViewGroup parent) {
+        ViewHolderItem viewHolderItem;
+		if (convertView == null) {
+			convertView = LayoutInflater.from(getContext()).inflate(R.layout.subforsublist, parent, false);
+			viewHolderItem = new ViewHolderItem();
+			viewHolderItem.t = ((TextView) convertView.findViewById(R.id.name));
```

Comment:
This should be refactored into the `ViewHolderItem` constructor call, not a separate assignment.